src/bot/commands/approve.ts

```diff
@@ -115,121 +178,124 @@ export default class ApproveCommand extends Command {
         where: {
-          id: proposal.id
+          id: proposal.id!
         }
       });
-      return interaction.reply(interactionProblem(`Le message est invalide.`));
+      await interaction.reply(interactionProblem(`Le message est invalide.`));
+      return null;
     }
 
     if (proposal.user_id === interaction.user.id) {
-      return interaction.reply(
-        interactionProblem(`Vous ne pouvez pas approuver votre propre ${Declaration[channel.id].WORD}.`)
-      );
-    }
-
-    if (proposal.merged) {
-      if (!embed.footer) {
-        embed.color = Colors.ACCEPTED;
-        embed.footer = { text: `${Declaration[channel.id].WORD_CAPITALIZED} déjà traité${isReport ? '' : 'e'}` };
-
-        const field = embed.fields?.[embed.fields.length - 1];
-        if (field) {
-          field.value = field.value.match(dataSplitRegex)!.groups!.base;
-        } else {
-          embed.description = embed.description!.match(dataSplitRegex)!.groups!.base;
-        }
-
-        await message.edit({ embeds: [embed] });
-      }
-
-      return interaction.reply(
-        interactionProblem(`${Declaration.EMBED_FOOTER_WITH_DETERMINANT} a déjà été ajouté${isReport ? '' : 'e'}.`)
+      await interaction.reply(
+        interactionProblem(`Vous ne pouvez pas approuver votre propre ${Declaration[message.channel.id].WORD}.`)
       );
+      return null;
     }
 
-    if (proposal.refused) {
-      if (!embed.footer) {
-        embed.color = Colors.REFUSED;
-        embed.footer = { text: `${Declaration[channel.id].WORD_CAPITALIZED} refusé${isReport ? '' : 'e'}` };
-
-        const field = embed.fields?.[embed.fields.length - 1];
-        if (field) {
-          field.value = field.value.match(dataSplitRegex)!.groups!.base;
-        } else {
-          embed.description = embed.description!.match(dataSplitRegex)!.groups!.base;
-        }
-
-        await message.edit({ embeds: [embed] });
-      }
+    const check = await checkProposalsifMergeOrRefused(interaction, proposal, message);
 
-      return interaction.reply(
-        interactionProblem(
-          `${Declaration[channel.id].WITH_DEMONSTRATIVE_DETERMINANT} a déjà été refusé${isReport ? '' : 'e'}.`
-        )
-      );
-    }
+    if (check) return null;
 
     if (isSuggestion) {
       const correction = proposal.corrections[0];
       if (correction) {
         const beenApproved = correction.approvals.some((approval) => approval.user_id === interaction.user.id);
         if (!beenApproved) {
-          const correctionLink = messageLink(interaction.guild.id, correctionsChannelId, correction.message_id!);
-          const suggestionLink = messageLink(interaction.guild.id, suggestionsChannelId, proposal.message_id!);
-          return interaction.reply(
+          const correctionLink = messageLink(interaction.guild!.id, correctionsChannelId, correction.message_id!);
+          const suggestionLink = messageLink(interaction.guild!.id, suggestionsChannelId, proposal.message_id!);
```

Comment:
Etant donné que le typage 'cached' est bon, guild n'est normalement pas optionel
```suggestion
          const correctionLink = messageLink(interaction.guild.id, correctionsChannelId, correction.message_id!);
          const suggestionLink = messageLink(interaction.guild.id, suggestionsChannelId, proposal.message_id!);
```